### PR TITLE
GUI overhaul + additional methods added in python scripts

### PR DESCRIPTION
**Pull request by ramblecube**
## GUI overhaul

The GUI changes include:

*   Removing all existing templates, and
*   Replacing them with new templates.

The detailed explanations for each new page are in their respective commits. Accompanied in parenthesis with their title bar texts in the attached screenshots, the new pages are:

*   splash.j2 (N/A)
*   landing.j2 ("Landing")
*   donate.j2 ("Donate")
*   browse.j2 ("Browse")
*   keyword.j2 ("Search by Keyword")
*   category.j2 ("Search by Category")
*   search.j2 ("Search Results")
*   app.j2 ("Application")

Each new page is based off of the "base template", base.j2.

### Script additions

These GUI changes rely on the following scripts:

*   static/js/common.js (common shop functions & download function)
*   static/js/buttons.js (button aggregation & mouse listeners)
*   static/js/tinycolor.js (the TinyColor JS library, [https://github.com/bgrins/TinyColor](https://github.com/bgrins/TinyColor) )
*   jquery 1.3.2 (the last jQuery version compatible with Opera 9)

### "Static" directory reorganization

The "static" directory has been reorganized into three subdirectories, those being "js", "img", and "css".

### Changes in app.py

In order to accommodate for the new pages, page routes have been added in app.py.

## Python script changes

Additional methods & dependencies have been added to app.py and osc.py.

In short, these additional methods allow for the following:

*   Formatting file-sizes (in bytes) into human-readable file-sizes (as in kb, mb, etc.)
*   Obtaining the app of the day from the oscwii.org website (and caching as appropriate)
*   Obtaining a "message of the day" from data/motd.txt
*   Searching the package list by partial keywords

## More details

More details on each change are provided in their respective commits.  
  
If this pull request, or any of the commits, are not in the expected format, or if any of the same violate Git etiquette; thank you for your patience, this is my first pull request.

### CODE REVIEW

**Review by dhtdht020 — CHANGES_REQUESTED** [review]

File: app.py [hunk 1]
```diff
@@ -23,68 +50,97 @@ def modify_query(**new_values):
...
+    return render_template('error.j2', errorcode=errorcode, errortext=errortext)
 
 
 if __name__ == '__main__':
     WSGIRequestHandler.protocol_version = "HTTP/1.1"
-    context = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
-    context.load_cert_chain('cert.pem', 'key.pem')
-
-    app.run(host='127.0.0.1', port=443, debug=True, ssl_context=context)
+    app.run(host='0.0.0.0', port=80, debug=True)
```
[the remark this answers is not in the document]
Comment (ramblecube) [reply app.py:146]: No, that was just for testing, thanks for catching that. I would be happy to set that back up.  
  
That has got me thinking, though- implementing SSL would require patching our root certificate into the channel, which is quite a few more bytes than is necessary for functionality witth plain HTTP. It is absolutely possible, though, and people like @spotlightishere have already done so, considering the working key + cert you sent me earlier.  
  
Considering secure information is not sent between the client and the server, however, I'm not yet aware of the advantages it would give us. With two bytes patched, the shop channel will trust all domains & HTTP connections for JavaScript & Opera purposes:



This is, ultimately, up to everyone, and if I am missing an advantage of HTTPS, or if HTTPS is just a preference for privacy purposes, whatever works works!

File: app.py [hunk 2]
```diff
@@ -23,68 +50,97 @@ def modify_query(**new_values):
 
 
 @app.route("/")
-def welcome():
-    return render_template('welcome.html')
+def splash():
+    return render_template('splash.j2')
```
[the remark this answers is not in the document]
Comment (ramblecube) [reply app.py]: Sounds good. Those have just been changed.

File: app.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dhtdht020) [reply app.py:146]: @spotlightishere @SketchMaster2001 @dakrk

**Comment by dhtdht020** [comment]
I made a few changes to the code, fixing an issue I found with the file size going through `sizeof_fmt` twice causing a type error, as well as a few refactorings to coding style.

I've noticed that all buttons around the new frontend don't seem to work for me on the Wii Shop, I would appreciate if you can look into it.

**Comment by ramblecube** [comment]
> I made a few changes to the code, fixing an issue I found with the file size going through `sizeof_fmt` twice causing a type error, as well as a few refactorings to coding style.
> 
> I've noticed that all buttons around the new frontend don't seem to work for me on the Wii Shop, I would appreciate if you can look into it.

Gah! I should have caught the stray `*.j2` references; nice catch. The `sizeof_fmt` fix is appreciated as well. I figure I should test code before commits, no?  
  
I'm not able to reproduce the buttons problem you're having- which patches are you using?

**Comment by dhtdht020** [comment]
> I'm not able to reproduce the buttons problem you're having- which patches are you using?
The patch in the group DM.

**Comment by ramblecube** [comment]
@dhtdht020 , I am still unable to run the program with the SSL context, but I suspect that the jQuery CDN is not whitelisted in the patch you mentioned. I have added a local copy of the appropriate jQuery to the `static/js` directory.

**Comment by dhtdht020** [comment]
Oh good! It's generally important to have every library included and the entire project self contained.. for reliability and alike.

**Review by dhtdht020 — APPROVED** [review]
